spec/features/two_factor_authentication/backup_code_sign_up_spec.rb

```diff
@@ -0,0 +1,31 @@
+require 'rails_helper'
+
+feature 'sign up with backup code', :js do
+  it 'works' do
+    sign_up_and_set_password
+    select_2fa_option('backup_code')
+    click_on 'Continue'
+  end
+
+  it 'works on signin with code' do
+    user = create(:user, :signed_up)
+    codes = BackupCodeGenerator.new(user).generate
+    signin(user.email, user.password)
+    fill_in :backup_code_verification_form_backup_code, with: codes.first
+    click_on 'Submit'
```

Comment:
Just because we fill in the code and click submit doesn't mean it was successful. What if the app returns an error after the submit?